packages/wrangler/src/autoconfig/run.ts

```diff
@@ -66,57 +66,55 @@ export async function runAutoConfig(
 		},
 	} satisfies RawConfig;
 
-	const { confirmed, modifications } = await buildAndConfirmOperationsSummary(
-		autoConfigDetails,
-		baseWranglerConfig
-	);
-
-	if (!confirmed) {
+	const modifications = await buildOperationsSummary(autoConfigDetails, {
+		...wranglerConfig,
+		...(await autoConfigDetails.framework?.configure({
+			outputDir: autoConfigDetails.outputDir,
+			projectPath: autoConfigDetails.projectPath,
+			workerName: autoConfigDetails.workerName,
+			dryRun: true,
+		})),
+	});
+
+	if (!(skipConfirmation || (await confirm("Proceed with setup?")))) {
 		throw new FatalError("Deployment aborted");
 	}
 
 	logger.debug(
 		`Running autoconfig with:\n${JSON.stringify(autoConfigDetails, null, 2)}...`
 	);
 
-	startSection("Configuring your application for Cloudflare");
-
 	if (modifications.wranglerInstall) {
 		await installWrangler();
 	}
 
-	if (modifications.typegenScriptAddition) {
-		assert(autoConfigDetails.packageJson);
+	if (autoConfigDetails.packageJson) {
 		await writeFile(
 			resolve(autoConfigDetails.projectPath, "package.json"),
 			JSON.stringify(
 				{
 					...autoConfigDetails.packageJson,
 					scripts: {
 						...autoConfigDetails.packageJson.scripts,
-						[typeGenScript.key]: typeGenScript.value,
+						...modifications.scripts,
 					},
 				},
 				null,
 				2
 			)
 		);
 	}
-
 	const additionalConfigDetails =
-		(await autoConfigDetails.framework?.configure(
-			autoConfigDetails.outputDir
-		)) ?? {};
+		(await autoConfigDetails.framework?.configure({
+			outputDir: autoConfigDetails.outputDir,
+			projectPath: autoConfigDetails.projectPath,
+			workerName: autoConfigDetails.workerName,
+			dryRun: false,
```

Comment:
For dry-run I was thinking that stopping at the summary step made sense (https://github.com/cloudflare/workers-sdk/pull/11258), since at that point we would already have presented all the useful information to the user, do you disagree?

Reply:
Expanding dry run to this stage means that we also tell users about the `wrangler.jsonc` modifications, scripts we'll add to their package.json, and installing Wrangler.